src/components/Chat.tsx

```diff
@@ -75,6 +73,20 @@ const Chat: React.FC<ChatProps> = ({
   const [inputMessage, setInputMessage] = useState("");
   const [abortController, setAbortController] = useState<AbortController | null>(null);
   const [loading, setLoading] = useState(false);
+  const [chatIsVisible, setChatIsVisible] = useState(false);
+
+  const visibleCallback = useCallback(() => {
+    emitter.addEventListener(
+      EVENT_NAMES.CHAT_IS_VISIBLE,
+      (evt: CustomEvent<{ chatIsVisible: boolean }>) => {
+        setChatIsVisible(evt.detail.chatIsVisible);
+      }
+    );
+  }, []);
+
+  useEffect(() => {
+    visibleCallback();
+  }, []);
```

Comment:
The event listener for EVENT_NAMES.CHAT_IS_VISIBLE is added in the useEffect hook but is never removed. This can lead to memory leaks if the Chat component is unmounted and remounted multiple times, as the event listener will persist and accumulate.